Mathlib/Algebra/Algebra/Basic.lean

```diff
@@ -132,18 +132,18 @@ end CommSemiring
 
 section Ring
 
-variable [CommRing R]
-variable (R)
-
 /-- A `Semiring` that is an `Algebra` over a commutative ring carries a natural `Ring` structure.
 See note [reducible non-instances]. -/
-abbrev semiringToRing [Semiring A] [Algebra R A] : Ring A :=
+abbrev semiringToRing (R : Type*) [CommRing R] [Semiring A] [Algebra R A] : Ring A :=
   { __ := (inferInstance : Semiring A)
     __ := Module.addCommMonoidToAddCommGroup R
     intCast := fun z => algebraMap R A z
     intCast_ofNat := fun z => by simp only [Int.cast_natCast, map_natCast]
     intCast_negSucc := fun z => by simp }
 
+instance {R : Type*} [Ring R] : Algebra (Subring.center R) R :=
+  Algebra.ofModule smul_mul_assoc mul_smul_comm
+
```

Comment:
I think maybe this is better?
```lean
instance {R : Type*} [Ring R] : Algebra (Subring.center R) R where
  toFun := Subtype.val
  map_one' := rfl
  map_mul' _ _ := rfl
  map_zero' := rfl
  map_add' _ _ := rfl
  commutes' := fun ⟨_, h⟩ x ↦ (Subring.mem_center_iff.1 h x).symm
  smul_def' _ _ := rfl
```
Although it's longer, but when it converts to map it is definitionally equal to `Subtype.val` and which is injective. I think this will make some of the following proofs shorter.

Reply:
Indeed it might. Thanks